services/harmony/app/util/cmr.ts

```diff
@@ -749,15 +773,183 @@ async function queryCollections(
  * @param token - Access token for user request
  * @returns The umm collection search results
  */
-async function queryUmmCollections(
+async function _queryUmmCollections(
   context: RequestContext, query: CmrQuery, token: string,
 ): Promise<Array<CmrUmmCollection>> {
+  logger.debug('Calling CMR to fetch UMM collections');
   const ummResponse = await _cmrGet(context, '/search/collections.umm_json_v1_17_3', query, token) as CmrUmmCollectionsResponse;
   return ummResponse.data.items;
 }
 
 /**
- * Performs a CMR grids.umm_json search with the given query string
+ * Generates a consistent MD5 hash for a given CMR query object and token.
+ *
+ * This function:
+ * - Sorts the keys of the query object to ensure consistent ordering.
+ * - Serializes the sorted query and token into a single JSON string.
+ * - Computes the MD5 hash of that string.
+ *
+ * This guarantees that the same query (regardless of key order)
+ * and the same token will always yield the same hash value.
+ *
+ * @param type - The CMR query type, e.g. 'json', 'umm_json'
+ * @param query - The CMR query object to hash.
+ * @param token - A string token to include in the hash computation.
+ * @returns A hexadecimal MD5 hash string representing the input.
+ */
+export function hashCmrQuery(type: string, query: CmrQuery, token: string): string {
```

Comment:
probably should write a unit test for this function

Reply:
unit test added.